packages/odyssey-react-mui/src/ui-shell/OktaUiShell.tsx

```diff
@@ -0,0 +1,123 @@
+/*!
+ * Copyright (c) 2024-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, useEffect, useState, type ReactNode } from "react";
+import { ErrorBoundary } from "react-error-boundary";
+
+import { SideNav, type SideNavProps } from "../labs/SideNav";
+import { TopNav, type TopNavProps } from "../labs/TopNav";
+import { OdysseyProvider } from "../OdysseyProvider";
+import { ShadowDomElements } from "../web-component/shadow-dom";
+
+const containerStyles = {
+  display: "flex",
+};
+
+export type OktaUiShellComponentProps = {
+  sideNavProps: Omit<SideNavProps, "logo" | "footerComponent">;
+  topNavProps: Omit<
+    TopNavProps,
+    "AdditionalNavItemComponent" | "SearchFieldComponent"
+  >;
+};
+
+export const defaultComponentProps: OktaUiShellComponentProps = {
+  sideNavProps: {
+    navHeaderText: "",
+    sideNavItems: [],
+  },
+  topNavProps: {
+    topNavLinkItems: [],
+  },
+};
+
+export type OktaUiShellProps = {
+  appComponent: ReactNode;
+  onError?: () => void;
+  onSubscriptionCreated: () => void;
+  optionalComponents?: {
+    additionalTopNavItems?: TopNavProps["AdditionalNavItemComponent"];
+    footer?: SideNavProps["footerComponent"];
+    logo?: SideNavProps["logo"];
+    searchField?: TopNavProps["SearchFieldComponent"];
+  };
+  subscribeToPropChanges: (
+    subscription: (componentProps: OktaUiShellComponentProps) => void,
+  ) => () => void;
+} & ShadowDomElements;
+
+const OktaUiShell = ({
+  appComponent,
+  appRootElement,
+  emotionRootElement,
+  onError = console.error,
+  onSubscriptionCreated,
+  optionalComponents,
+  subscribeToPropChanges,
+}: OktaUiShellProps) => {
+  const [componentProps, setComponentProps] = useState(defaultComponentProps);
+
+  useEffect(() => {
+    const unsubscribe = subscribeToPropChanges((componentProps) => {
+      setComponentProps(componentProps);
+    });
+
+    onSubscriptionCreated();
+
+    return () => {
+      unsubscribe();
+    };
+  }, [onSubscriptionCreated, subscribeToPropChanges]);
+
+  return (
+    <ErrorBoundary fallback={appComponent} onError={onError}>
```

Comment:
This actually only accounts for the `OdysseyProvider` failing. Since we're passing values to `OktaUiShell` itself, those errors aren't caught here.

I had to wrap `OktaUiShell` in _another_ `ErrorBoundary` to make this work.

Ideally, I'd also have it setup to re-render if you update the props in that state setter function, but that's not possible in the current implementation.